packages/core/src/fhir-deduplication/resources/condition.ts

```diff
@@ -50,19 +63,40 @@ export function groupSameConditions(conditions: Condition[]): {
   }
 
   for (const condition of conditions) {
+    if (hasBlacklistedText(condition.code)) {
+      danglingReferencesSet.add(createRef(condition));
+      continue;
+    }
+
     const date = getDateFromResource(condition);
-    const { snomedCode, icd10Code } = extractCodes(condition.code);
+    if (!date) {
+      danglingReferencesSet.add(createRef(condition));
+      continue;
+    }
 
-    if (icd10Code && date) {
-      const compKey = JSON.stringify(createCompositeKey(icd10Code, date));
+    const { snomedCode, icd10Code } = extractCodes(condition.code);
+    if (icd10Code) {
+      const compKey = JSON.stringify({ icd10Code, date });
       fillMaps(icd10Map, compKey, condition, refReplacementMap, undefined, removeOtherCodes);
-    } else if (snomedCode && date) {
-      const compKey = JSON.stringify(createCompositeKey(snomedCode, date));
+    } else if (snomedCode) {
+      const compKey = JSON.stringify({ snomedCode, date });
       fillMaps(snomedMap, compKey, condition, refReplacementMap, undefined, removeOtherCodes);
+    } else {
+      const display = extractDisplayFromConcept(condition.code);
+      if (display) {
+        const compKey = JSON.stringify({ display, date });
```

Comment:
should we call this a backupKey instead?

Reply:
i dont see the point of renaming it. I think it might confuse us later on. The purpose of `key` vs `backupKey` is exactly the same